charts/gitops-server/templates/deployment.yaml

```diff
@@ -63,6 +63,30 @@ spec:
               port: http
           {{- if or .Values.enableLogin .Values.envVars}}
           env:
+            - name: PROFILE_CACHE_LOCATION
+              value: {{ .Values.profileCacheLocation }}
+            - name: HEALTHZ_BIND_ADDRESS
+              value: {{ .Values.watcherHealthzBindAddress }}
+            - name: METRICS_BIND_ADDRESS
+              value: {{ .Values.watcherMetricsBindAddress }}
+            - name: NOTIFICATION_CONTROLLER_ADDRESS
+              value: {{ .Values.notificationControllerAddress }}
+            - name: WATCHER_PORT
+              value: {{ .Values.watcherPort }}
+            - name: OIDC_ISSUER_URL
+              value: {{ .Values.OIDCIssuerURL }}
+            - name: OIDC_CLIENT_ID
+              value: {{ .Values.OIDCClientID }}
+            - name: OIDC_CLIENT_SECRET
```

Comment:
After looking at this further, I'm not sure it makes sense to include these OIDC in the template because the server works in two modes:
* Read a secret from the cluster that contains all important OIDC data
* Take individual OIDC CLI parameter including a client "secret"

If we were trying to support the latter model, we would need to have a user create a secret in the cluster just so it could be referenced in the creation of the actual OIDC secret. When the server was run from the command line, the extra secret was unnecessary.

Reply:
I think I'm confused by what you mean here? 

I thought `OIDC_CLIENT_SECRET` was to pass the secret information that is used to identify the app to the OIDC provider. This should definitely be referenced from a k8s `secret`. Which we can create via helm. 

It's useful to also support the option for the user to create the secret via an alternate method (e.g. using SOPS + flux) so we'll probably need to add flag like `oidc.secretName` which, if not set, means we create a secret (& expect to be passed a value for client_secret). e.g.:
```yaml
# deployment.yaml 
#...
- name: OIDC_CLIENT_SECRET
  valueFrom:
    secretKeyRef: 
      name: {{ default "weave-gitops-app-oidc-client-secret" .Values.oidc.secretName }}
      key: oidc_client_secret
#...
# oidc-secret.yaml
{{- if not .Values.oidc.secretName }}
kind: Secret
apiVersion: v1
metadata:
  name: weave-gitops-app-oidc-client-secret
data:
  oidc_client_secret: {{ .Values.oidc.clientSecret | required "client secret must be set if secretName isn't" | b64enc }}
{{ end }}
```

This means that either a user provides us with a secret name for their OIDC client secret (which we'll require be created in the same namespace) and we use that or they provide us with the client secret itself which we'll use to create a secret for them. 

Or am I missing something?

Reply:
What I meant was that in the chart case, our only option for setting this up is to use a secret. In the code, we either use a client secret provided by the user or _another_ secret containing the OIDC_CLIENT_SECRET combined with the other required parameters. I don't see how it helps the user to create a secret only containing the OIDC_CLIENT_SECRET and then set up the remaining parameters via the values file. If the user is going to create a secret anyway, why not just create the client secret itself?

Reply:
ah right, yes creating the whole client secret makes sense